test/functional/lua/vim_spec.lua

```diff
@@ -2155,6 +2155,27 @@ describe('lua stdlib', function()
     ]]
 
     eq(2, funcs.luaeval "vim.g.var")
+
+    -- nvim_cmd can handle 0 ranges (#19608)
+    local screen = Screen.new(60,5)
+    screen:attach()
+    exec_lua [[
+    vim.api.nvim_buf_set_lines(0, 0, -1, false, { "aa" })
+    ]]
+    exec_lua [[
+    vim.cmd.delete{ range = { 0 } }
+    vim.cmd.undo()
+    ]]
+    screen:expect{grid=[[
+      ^aa                                                          |
+      {1:[No Name] [+]                                               }|
+                                                                  |
+      {2:[No Name]                                                   }|
+      1 more line; before #2  0 seconds ago                       |
```

Comment:
This is flaky. It can sometimes be `1 second ago` here. I also think this test fits better in `api/vim_spec.lua`. I wrote a test that fails without this fix:
```diff
diff --git a/test/functional/api/vim_spec.lua b/test/functional/api/vim_spec.lua
index 17de6730f..dfa10103c 100644
--- a/test/functional/api/vim_spec.lua
+++ b/test/functional/api/vim_spec.lua
@@ -3822,5 +3822,11 @@ describe('API', function()
       feed("[l")
       neq(nil, string.find(eval("v:errmsg"), "E5108:"))
     end)
+    it('handles 0 range #19608', function()
+      meths.buf_set_lines(0, 0, -1, false, { "aa" })
+      meths.cmd({ cmd = 'delete', range = { 0 } }, {})
+      command('undo')
+      assert_alive()
+    end)
   end)
 end)
```

Reply:
This test doesn't consistently fail for me.

Reply:
Adding `eq({'aa'}, meths.buf_get_lines(0, 0, 1, false))` makes it consistently fail.